packages/backend/src/open_payments/auth/grant.ts

```diff
@@ -108,3 +142,58 @@ export class Grant {
     }
   }
 }
+
+// Export for testing
+export function getInterval(
+  repeatingInterval: string,
+  target: Date
+): Interval | undefined {
+  const parts = repeatingInterval.split('/')
+  assert.ok(parts.length === 3)
+
+  let repetitions: number | undefined
+  if (parts[0].length > 1 && parts[0][1] !== '-') {
+    repetitions = Number(parts[0].slice(1))
+  } else if (
+    (parts[0].length === 1 && parts[0][0] === 'R') ||
+    (parts[0].length > 1 && parts[0][1] === '-')
+  ) {
+    repetitions = Infinity
+  }
+  if (repetitions === undefined || isNaN(repetitions)) return
+
+  const interval = Interval.fromISO(`${parts[1]}/${parts[2]}`)
+  if (!interval.start) return
+  if (interval.contains(DateTime.fromJSDate(target))) return interval
+
+  let duration: string
+  let forward: boolean
+  if (parts[1].length > 1 && parts[1][0] === 'P') {
+    duration = parts[1]
+    forward = false
+  } else if (parts[2].length > 1 && parts[2][0] === 'P') {
+    duration = parts[2]
+    forward = true
+  } else {
+    duration = interval.toDuration().toString()
+    forward = true
+  }
```

Comment:
These checks were already performed inside [Interval.fromISO](https://github.com/moment/luxon/blob/0cbffd9c87eca6ac9a75dce124ce890dd945bc78/src/interval.js#L134)
What if we call `DateTime.fromISO`/`Duration.fromISO`/etc on `parts[1]` and `parts[2]` instead of `Interval.fromISO`, so that we can capture `forward` and the duration (as a `Duration`, and not re-parse the duration for every interval below)?

Reply:
If don't use `Interval.fromISO`, we can't capture those intervals of the form `repetition/start/end`. But I can move the parsing out of the loop.

Reply:
>If don't use `Interval.fromISO`, we can't capture those intervals of the form `repetition/start/end`.

Not sure I follow.
I think we should be able to replace the call to `Interval.fromISO` by implementing this part of it ourselves (still using luxon to `DateTime.fromISO` and `Duration.fromISO`):
https://github.com/moment/luxon/blob/0cbffd9c87eca6ac9a75dce124ce890dd945bc78/src/interval.js#L137-L167